File: playgrounds/mongoose/app/models/book.js
'use strict';

const mongoose = require('mongoose');

const { Schema } = mongoose;

const bookSchema = new Schema(
  {
    title: { type: String, required: true, trim: true },
    author: { type: String, required: true },
    genres: { type: [String], default: [] },
    pages: { type: Number, min: 1 },
    published: { type: Date },
    inStock: { type: Boolean, default: true },
    rating: { type: Number, min: 0, max: 5 },
  },
  { timestamps: true }
);

// Compound index: exercises DocumentDB index creation via Mongoose autoIndex.
// Note: no `collation` option here; DocumentDB does not implement collation indexes.
bookSchema.index({ author: 1, title: 1 });

module.exports = mongoose.model('Book', bookSchema);

File: playgrounds/beanie/app/models/book.py
"""Example Beanie document model used by the demo API.

Mirrors the Mongoose ``Book`` schema in the sibling playground so the two demos
are directly comparable. Note there is intentionally **no index collation**:
DocumentDB does not implement collation indexes.
"""

from __future__ import annotations

from datetime import datetime, timezone
from typing import List, Optional

from beanie import Document
from pydantic import Field
from pymongo import ASCENDING, IndexModel


def _utcnow() -> datetime:
    return datetime.now(timezone.utc)


class Book(Document):
    title: str
    author: str
    genres: List[str] = Field(default_factory=list)
    pages: Optional[int] = None
    published: Optional[datetime] = None
    in_stock: bool = True
    rating: Optional[float] = None
    created_at: datetime = Field(default_factory=_utcnow)
    updated_at: datetime = Field(default_factory=_utcnow)

    class Settings:
        name = "books"
        # Compound index: exercises DocumentDB index creation via Beanie.
        # No `collation` option here; DocumentDB does not implement it.
        indexes = [
            IndexModel([("author", ASCENDING), ("title", ASCENDING)]),
        ]
